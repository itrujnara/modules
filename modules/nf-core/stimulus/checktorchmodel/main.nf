process STIMULUS_CHECKTORCHMODEL {
    tag "$experiment_config - $original_csv"
    label 'process_medium'

    // TODO freeze to Wave
    container "docker.io/mathysgrapotte/stimulus-py:latest"

    input:
    path(original_csv)
    path(experiment_config)
    path(model)
    path(ray_tune_config)
    path(initial_weights)

    output:
    tuple val(meta), path("*_modelcheck.log"), emit: log
    path "versions.yml"                      , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: model.replaceFirst(/\.py/, "")
    """
    stimulus-check-model \
        -d ${original_csv} \
        -m ${model} \
        -e ${experiment_config} \
        -c ${ray_tune_config} \
        --initial_weights ${initial_weights} \
        --gpus ${task.accelerator.request} \
        --cpus ${task.cpus} \
        --memory "${task.memory}" \
        $args > ${prefix}_modelcheck.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        Python: \$(python --version | cut -d ' ' -f 2)
        Stimulus-py: \$(pip show stimulus-py | grep Version | cut -d ' ' -f 2)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def STIMULUS_VER = '0.0.9' // container not used in stub, change manually
    """
    touch ${meta.id}_modelcheck.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        Python: \$(python --version | cut -d ' ' -f 2)
        Stimulus-py: ${STIMULUS_VER}
    END_VERSIONS
    """
}
