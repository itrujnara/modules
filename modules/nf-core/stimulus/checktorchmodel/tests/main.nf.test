// nf-core modules test stimulus/checktorchmodel
nextflow_process {

    name "Test Process STIMULUS_CHECKTORCHMODEL"
    script "../main.nf"
    process "STIMULUS_CHECKTORCHMODEL"

    tag "modules"
    tag "modules_nfcore"
    tag "stimulus"
    tag "stimulus/checktorchmodel"

    test("basic - csv - py") {

        when {
            process {
                """
                input[0] = file(params.modules_testdata_base_path + '../../deepmodeloptim/testdata/titanic/titanic_stimulus.csv', checkIfExists: true)
                input[1] = file(params.modules_testdata_base_path + '../../deepmodeloptim/testdata/titanic/titanic.yaml', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + '../../deepmodeloptim/testdata/titanic/titanic_model.py', checkIfExists: true)
                input[3] = file(params.modules_testdata_base_path + '../../deepmodeloptim/testdata/titanic/titanic_model_cpu.yaml', checkIfExists: true)
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { with(process.out.log) {
                    assert path(get(1)).readLines().first().contains("Trial finishes successfully")
                } },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("basic - csv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = file(params.modules_testdata_base_path + '../../deepmodeloptim/testdata/titanic/titanic_stimulus.csv', checkIfExists: true)
                input[1] = file(params.modules_testdata_base_path + '../../deepmodeloptim/testdata/titanic/titanic.yaml', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + '../../deepmodeloptim/testdata/titanic/titanic_model.py', checkIfExists: true)
                input[3] = file(params.modules_testdata_base_path + '../../deepmodeloptim/testdata/titanic/titanic_model_cpu.yaml', checkIfExists: true)
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
