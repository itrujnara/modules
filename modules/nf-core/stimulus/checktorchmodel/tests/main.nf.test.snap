{
    "basic - csv - py": {
        "content": [
            [
                "versions.yml:md5,84d45b8dcf715e8f42c82339832c0f29"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-11T16:24:18.958811"
    },
    "basic - csv - stub": {
        "content": [
            {
                "0": [
                    "titanic_model_modelcheck.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "versions.yml:md5,70fcb1abbe6e6a2f1dad708b98db1843"
                ],
                "log": [
                    "titanic_model_modelcheck.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,70fcb1abbe6e6a2f1dad708b98db1843"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-11T16:24:24.093869"
    }
}